portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,13 +40,26 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String maxContent = request.getParameter("max");
+    int maxComments;
+    try {
+      maxComments = Integer.parseInt(maxContent);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxContent);
+      return;
+    }
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     List<String> messages = new ArrayList<>();
+    int numComments = 0;
```

Comment:
Can you check the size of messages list instead?

Reply:
Oh that's embarrassing 🤣